otherlibs/dune-rpc/private/exported_types.ml

```diff
@@ -43,6 +44,26 @@ module Target = struct
     | Preprocess of string list
     | Loc of Loc.t
 
+  let compare x y =
```

Comment:
You can just use a map keyed by the diagnostic id's

Reply:
Ofc!